modules/product/src/main/java/com/opengamma/strata/product/etd/EtdIdUtils.java

```diff
@@ -0,0 +1,160 @@
+    ArgChecker.notNull(exchangeId, "exchangeId");
+    ArgChecker.notEmpty(contractCode, "contractCode");
+    switch (type) {
+      case FUTURE:
+        return EtdContractSpecId.of(ETD_SCHEME, FUT_PREFIX + exchangeId + "/" + contractCode);
+      case OPTION:
+        return EtdContractSpecId.of(ETD_SCHEME, OPT_PREFIX + exchangeId + "/" + contractCode);
+      default:
+        throw new IllegalArgumentException("Unknown ETD type: " + type);
+    }
+  }
+
+  /**
+   * Creates an identifier for an ETD future instrument.
+   * <p>
+   * A typical monthly ETD will have the format:
+   * {@code 'OG-Etd~O/ECAG/OGBS/201706'}.
+   * <p>
+   * A more complex flex ETD (12th of the month, Physical settlement) will have the format:
+   * {@code 'OG-Etd~O/ECAG/OGBS/20170612E'}.
+   *
+   * @param exchangeId  the MIC code of the exchange where the instruments are traded
+   * @param contractCode  the code supplied by the exchange for use in clearing and margining, such as in SPAN
+   * @param expiryMonth  the month of expiry
+   * @param style  the style of the ETD, such as 'Monthly', 'Weekly, 'Daily' or 'Flex.
+   * @return the identifier
+   */
+  public static SecurityId futureId(
+      ExchangeId exchangeId,
+      String contractCode,
+      YearMonth expiryMonth,
+      EtdStyle style) {
+
+    ArgChecker.notNull(exchangeId, "exchangeId");
+    ArgChecker.notEmpty(contractCode, "contractCode");
+    ArgChecker.isTrue(expiryMonth.getYear() >= 1000 && expiryMonth.getYear() <= 9999, "Invalid expiry year: ", expiryMonth);
+    ArgChecker.notNull(style, "style");
+
+    String id = FUT_PREFIX +
+        exchangeId + "/" +
+        contractCode + "/" +
+        expiryMonth.getYear() +
+        ((char) ((expiryMonth.getMonthValue() / 10) + '0')) +
+        ((char) ((expiryMonth.getMonthValue() % 10) + '0')) +
+        style.getCode();
+    return SecurityId.of(ETD_SCHEME, id);
+  }
+
+  /**
+   * Creates an identifier for an ETD future instrument.
+   * <p>
+   * A typical monthly ETD with version zero will have the format:
+   * {@code 'OG-Etd~O/ECAG/OGBS/201706/P1.50'}.
+   * <p>
+   * A more complex flex ETD (12th of the month, Cash settlement, European) with version two will have the format:
+   * {@code 'OG-Etd~O/ECAG/OGBS/20170612CE/V2/P1.50'}.
+   *
+   * @param exchangeId  the MIC code of the exchange where the instruments are traded
+   * @param contractCode  the code supplied by the exchange for use in clearing and margining, such as in SPAN
+   * @param expiryMonth  the month of expiry
+   * @param style  the style of the ETD, such as 'Monthly', 'Weekly, 'Daily' or 'Flex.
+   * @param version  the non-negative version, zero by default
+   * @param putCall  the Put/Call flag
+   * @param strikePrice  the strike price
+   * @return the identifier
+   */
+  public static SecurityId optionId(
+      ExchangeId exchangeId,
+      String contractCode,
+      YearMonth expiryMonth,
+      EtdStyle style,
+      int version,
+      PutCall putCall,
```

Comment:
There is no null check for this. If it's null the method will silently return an ID for a call option which probably isn't what we want.